framework/Util/Point.swift

```diff
@@ -1,9 +1,10 @@
 // class defining a point
 public struct Point {
-  public var x : Float
-  public var y : Float
+  public var x : Float = 0
+  public var y : Float = 0
   public init(_ x : Float, _ y : Float){
     self.x = x
     self.y = y
   }
+  public init() {}
```

Comment:
I'd remove the blank initializer for zero, in favor of something like this:

```
public static let zero = Point(0.0, 0.0)
```

That way, you can call Point.zero for a 0, 0 point and that becomes clear from the code itself.